Iceberg-Metacello-Integration/IceGitLocalRepositoryType.class.st

```diff
@@ -61,18 +71,21 @@ IceGitLocalRepositoryType >> mcRepository [
 
 { #category : 'private' }
 IceGitLocalRepositoryType >> splitRootAndSubdirectoryFromLocation [
-	| root subDir |
-	(self location beginsWith: (self class type, '://'))
-		ifFalse: [ self error: 'Invalid URL (It should be ', self class type, '://...)' ].
-	root := (self location allButFirst: self class type size + 3) asFileReference.
+	| urlPrefix  subDir root |
+	urlPrefix := self class type, '://'.
 	subDir := #().
+
+	(self location beginsWith: urlPrefix)
+		ifFalse: [ self error: 'Invalid URL (It should be ', urlPrefix ].
+	root := self homoginizePath: urlPrefix.
+
 	[ root isRoot or: [ self isGitRoot: root ] ] 
 	whileFalse: [ 
 		(root asAbsolute = FileSystem workingDirectory)
 			ifFalse: [ subDir := subDir copyWithFirst: root basename ].
 		root := root parent ].
 	root isRoot ifTrue: [ self error: 'I can''t find a .git/config file.' ].
-	
+
```

Comment:
This is super strange.

The code you're changed should already support '..' in github paths.

See for example:

```
'..' asFileReference
```

Reply:
As I wrote above I would like to go to the parents directory.  As for the FileReference the reason was the `#isGitRoot` method.

I will demonstrate it on an example:

This is a working directory - which will change for every user (or any directory the user specifies in `#expandedParent`)
```
/home/<user>/<working_pharo_directory/
```

but e.g. libraries would be in different folders:
```
/home/<user>/important_stuff/libraries
```

so I would like to load the library like:
```
Metacello new
  baseline: 'VMMaker';
  repository: 'gitlocal://../important_stuff/libraries;
  load;
  lock.
```

Reply:
I just did the following to load a (fresh) mutalk clone (from here https://github.com/pharo-contributions/mutalk) and this worked perfectly...

```smalltalk
Metacello new
  baseline: 'MuTalk';
  repository: 'gitlocal://../mutalk';
  load.
```

If you're having an issue there should be some difference in configuration.

I'm tested on: 

Pharo-13.0.0+SNAPSHOT.build.593.sha.9aa55368d349ba11822b37239dc8129f160b5fbb (64 Bit)
Build date: 19 March 2025

Reply:
Sigh, according to my testing you are right.  For whatever reason it works now when I use fresh image.  I would like to understand why it did not work for me before...  Thank you for you patience.